### PR TITLE
[Review view] SDTM Phase 1: R conversion functions + tests

### PR DESCRIPTION
**Pull request by Gero1999**
## ⚠️ Review-only PR — do not merge

This is a **review view** into the SDTM changes from #1318. It isolates the R package conversion functions for focused code review. The actual merge will happen via #1318.

## Scope

New R functions in `R/sdtm_to_pknca.R` that convert CDISC SDTM domain data (PC, EX, DM) into `PKNCAdata` objects:

| Function | Purpose |
|---|---|
| `sdtm_to_PKNCAdata(pc, ex, dm, metabolites)` | Top-level entry point, returns `PKNCAdata` |
| `pc_to_PKNCAconc(pc, ex, dm)` | PC domain → `PKNCAconc` with timing derivations |
| `ex_to_PKNCAdose(ex)` | EX domain → `PKNCAdose` with duration/route parsing |
| `parse_iso8601_duration(x)` | ISO 8601 duration → numeric hours |
| `route_cdisc_to_pknca(routes)` | CDISC route terms → PKNCA values |
| `std_dtc_to_rdate(dtc)` | ISO 8601 datetime → POSIXct |

Plus internal helpers: `.check_required_cols()`, `.prepare_dose_table()`, `.assign_doses_to_conc()`.

## Files changed

- `R/sdtm_to_pknca.R` — 537 lines, all new
- `NAMESPACE` — 4 new exports
- `man/` — 9 new `.Rd` files
- `tests/testthat/test-sdtm_to_pknca.R` — 908 lines, 60+ tests

## What to review

- Correctness of timing variable derivations (AFRLT, ARRLT, NFRLT, NRRLT)
- Dose assignment logic (DOSNOA, ATPTREF)
- ISO 8601 parsing edge cases
- Route mapping completeness
- Test coverage

### CODE REVIEW

File: R/sdtm_to_pknca.R [hunk 1]
```diff
@@ -0,0 +1,537 @@
...
+    } else {
+      as.numeric(ex$EXDUR)
+    }
+  } else if ("EXENDTC" %in% names(ex)) {
+    end_posix <- std_dtc_to_rdate(ex$EXENDTC)
+    dur <- as.numeric(difftime(end_posix, exstdtc_posix, units = "hours"))
+    adosedur <- ifelse(is.na(dur), 0, dur)
+  } else {
+    adosedur <- rep(0, nrow(ex))
+  }
```
Comment (KOBANAK) [inline R/sdtm_to_pknca.R:107]: Should we modify the duration calculation logic to prevent crashes when EXDUR column is present but entirely empty (containing only NAs or blank strings). Like instead falling through to EXENDTC which might have actual values ? 
```suggestion
if ("EXDUR" %in% names(ex) && !all(is.na(ex$EXDUR) | !nzchar(ex$EXDUR))) {
  adosedur <- if (is.character(ex$EXDUR)) {
    parse_iso8601_duration(ex$EXDUR)
  } else {
    as.numeric(ex$EXDUR)
  }
} else if ("EXENDTC" %in% names(ex)) {
  end_posix <- std_dtc_to_rdate(ex$EXENDTC)
  adosedur <- as.numeric(difftime(end_posix, exstdtc_posix, units = "hours"))
} else {
  adosedur <- rep(0, nrow(ex))
}
adosedur <- ifelse(is.na(adosedur), 0, adosedur)
```
